pkg/k8s/client_test.go

```diff
@@ -205,12 +286,23 @@ func TestGetService(t *testing.T) {
 		},
 	}
 
+	serviceKeyFunc := func(obj interface{}) string {
+		svc := obj.(*corev1.Service)
+		return fmt.Sprintf("%s/%s", svc.GetNamespace(), svc.GetName())
+	}
+
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			a := assert.New(t)
-			c, err := newClient(testclient.NewSimpleClientset(), nil, testMeshName, nil, nil)
+			a := tassert.New(t)
+			serviceStore := map[string]interface{}{}
+			store := newFakeCacheStore(serviceStore, serviceKeyFunc)
+			stores := map[informers.InformerKey]cache.Store{
+				informers.InformerKeyService: store,
+			}
+			ic, err := informers.NewInformerCollection(testMeshName, nil, informers.WithKubeClient(testclient.NewSimpleClientset()), informers.WithCustomStores(stores), informers.WithSelectedInformers(k8sInformerKeys...))
 			a.Nil(err)
-			_ = c.informers[Services].GetStore().Add(tc.service)
+			c := newClient(ic, nil, nil)
+			_ = store.Add(tc.service)
```

Comment:
So the real store doesn't return the value immediately when doing an add followed by a get?

I thought the only time it wasn't immediate is when doing an add through the actual client, which triggers an async add.

I believe this direct add is synchronous

Reply:
Because we can't access the real store from this package

Reply:
I see. adding all of this extra stuff to get around this testing seems like a bit of a code smell IMO. I would recommend 1 of 3 things:

1) Add all the objects to the k8s client, ie `testclient.NewSimpleClientset(tc.namespace, ...)`, then use the internals to ensure that the informer `HasSynced`.
2) Expose an add method on the store itself, so you don't need this fake store.
3) Don't separate these into 2 different packages if you need to rely so heavily on internal implementation details